src/server.rs

```diff
@@ -23,6 +23,22 @@ impl Server {
         Self(zt)
     }
 
+    pub async fn bind(ip: IpAddr) -> Result<(TcpListener, UdpSocket), anyhow::Error> {
+        let sa = SocketAddr::new(ip, 53);
+
+        let tcp = match TcpListener::bind(sa).await {
```

Comment:
Instead of matching the error and swallowing it, it would make more sense to either re-throw the error with the `?` operator, or reformat it with `anyhow!()` so the user could actually see what the error was, instead of getting this error for situations where f.e. the port is already bound.